extensions/warp-ipfs/src/lib.rs

```diff
@@ -1658,7 +1658,7 @@ impl RayGunAttachment for WarpIpfs {
         conversation_id: Uuid,
         message_id: Uuid,
         file: &str,
-    ) -> Result<BoxStream<'static, Result<Vec<u8>, Error>>, Error> {
+    ) -> Result<(usize, BoxStream<'static, Result<Vec<u8>, Error>>), Error> {
```

Comment:
Instead of making it a tuple, we should probably create a struct that is a stream that would house that metadata while still serving the bytes itself.

Reply:
ye i can change that